flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/TranslationContext.scala

```diff
@@ -53,21 +55,26 @@ object TranslationContext {
     // configure sql parser
     // we use Java lex because back ticks are easier than double quotes in programming
     // and cases are preserved
-    val parserConfig = SqlParser.configBuilder().setLex(Lex.JAVA).build()
+    val parserConfig = SqlParser
+      .configBuilder()
+      .setLex(Lex.JAVA)
+      .build()
 
     // initialize RelBuilder
     frameworkConfig = Frameworks
       .newConfigBuilder
       .defaultSchema(tables)
       .parserConfig(parserConfig)
       .costFactory(new DataSetCostFactory)
-      .traitDefs(ConventionTraitDef.INSTANCE)
+      .programs(Programs.ofRules(FlinkRuleSets.DATASET_OPT_RULES))
```

Comment:
I think this line can be removed because we set the rules explicitly before calling the optimizer.